### DISCUSSION TITLE
How To Bind a Key to Sending a Custom Escape Code (Text Binding)?

### DISCUSSION THREAD

**Discussion by VehementHam** [kovidgoyal/kitty#7512] [Q&A]
I have custom keybindings in ZSH, that involve things like **Ctrl** + **Shift** + **\<letter\>**. Bindings like this do not send unique escape codes. To work around this, I define text bindings. Here is a an example for the foot terminal:
```ini
[text-bindings]
\x1b[1;2F = Shift+KP_End
\x1b[1;2H = Shift+KP_Home

\x1b[27;6;65~ = Control+Shift+a
\x1b[27;6;66~ = Control+Shift+b
#\x1b[27;6;67~ = Control+Shift+c
\x1b[27;6;68~ = Control+Shift+d
\x1b[27;6;69~ = Control+Shift+e
\x1b[27;6;70~ = Control+Shift+f
\x1b[27;6;71~ = Control+Shift+g
\x1b[27;6;72~ = Control+Shift+h
\x1b[27;6;73~ = Control+Shift+i
\x1b[27;6;74~ = Control+Shift+j
\x1b[27;6;75~ = Control+Shift+k
\x1b[27;6;76~ = Control+Shift+l
\x1b[27;6;77~ = Control+Shift+m
#\x1b[27;6;78~ = Control+Shift+n
#\x1b[27;6;79~ = Control+Shift+o
\x1b[27;6;80~ = Control+Shift+p
\x1b[27;6;81~ = Control+Shift+q
#\x1b[27;6;82~ = Control+Shift+r
\x1b[27;6;83~ = Control+Shift+s
\x1b[27;6;84~ = Control+Shift+t
#\x1b[27;6;85~ = Control+Shift+u
#\x1b[27;6;86~ = Control+Shift+v
\x1b[27;6;87~ = Control+Shift+w
\x1b[27;6;88~ = Control+Shift+x
#\x1b[27;6;89~ = Control+Shift+y
\x1b[27;6;90~ = Control+Shift+z
```
How do I do this in Kitty?

**Comment by kovidgoyal** [comment]
This is in the kitty FAQ.

**Reply by VehementHam** [reply]
I do not see it in the FAQ.

**Comment by kovidgoyal** [comment]
https://sw.kovidgoyal.net/kitty/faq/#how-do-i-map-key-presses-in-kitty-to-different-keys-in-the-terminal-program
 Read the actual text not just the heading.

**Reply by VehementHam** [reply]
Is it this?

> Which will print out what key events it receives. To send arbitrary text rather than a key press, see [send_text](https://sw.kovidgoyal.net/kitty/conf/#shortcut-kitty.Send-arbitrary-text-on-key-presses) instead.

**Comment by VehementHam** [comment]
Okay so I did this:
```
map ctrl+shift+a send_text ^[[27;6;65
```
But when I run `cat`, and press **Ctrl** + **Shift** + **a**, the escape code is `^[[97;6u`, instead of `^[[27;6;65`.

**Reply by kovidgoyal** [reply]
I suggest you actually read the documentation and understand it. You are missing the modes in which the text is to be sent and your syntax for specifying the bytes in incorrect.

**Reply by VehementHam** [reply]
I'm sorry, but what do you mean by "modes" and "bytes," to what documentation specifically, are you refering to?

**Comment by VehementHam** [comment]
Could you tell me what documentation I need to read?
